### ISSUE TITLE
[Config Support]: Can't get restream to work on one camera

### ISSUE THREAD

**Issue by marcgarciamarti** [blakeblackshear/frigate#8781]
### Describe the problem you are having

I have 2 different anpviz cameras set up as rtsp cameras in frigate. I managed to successfully display the rtsp video feed of both. 
I decided to enable go2rtc recently, as I wanted a more lightweight approach to displaying the videostreams on Home Assistant. After reading the documentation, I managed to set it up on one camera but not on the other one.
The one that is working  was originally set up to publish one main stream and one substream. The one that is not working well is configured to display 3 streams (main, Sub-stream, third stream). The reason for this is that when running detection on the sub-stream, the resolution offered by the sub-stream was low and the system would create false positives. I yesterday reached out to the vendor and they confirm the camera is able to handle 4 channels. 
As I thought I could be facing a resource starvation, I started considering leveraging restreaming the video feed. The reason why I create this issue is I'm unable to get this second camera to properly restream the video feed, while its sibling is working just fine.

For testing purposes I even switched off the sub-stream channel, before registering this issue but the problem still happens.

In a nutshell:
- The camera that I have managed to get working with webrtc is not doing restreaming
- The camera that does not work, despite being a very similar model (not the very same one) is not working without restreaming
- I can't get the camera that does not work to restream, in order to avoid resource starvation on the camera 

Does anybody know whether my configuration file is incorrectly configured? the problematic camera is the one with name anpviz-camera-dome-comedor. 

Port 8554 is enabled in the Frigate add-on in home assistant.

### Version

0.12

### Frigate config file

```yaml
mqtt:
  host: 192.168.1.5
  user: USER
  password: PASSWORD

birdseye:
  enabled: True
  mode: motion

go2rtc:
  streams:
    anpviz-camera-dome-comedor:
      - rtsp://192.168.1.31:554/stream0
      - "ffmpeg:anpviz-camera-dome-comedor#audio=opus"
    anpviz-camera-dome-comedor_sub:
      - rtsp://192.168.1.31:554/stream2 
      - "ffmpeg:anpviz-camera-dome-comedor_sub#audio=opus" 

    anpviz-camera-dome-recibidor:
      - rtsp://192.168.1.32:554/stream0
      
cameras:
  anpviz-camera-dome-comedor:
    ffmpeg:
      hwaccel_args: preset-intel-qsv-h264
      output_args:
        record: preset-record-generic-audio-copy

      inputs:
        - path: rtsp://127.0.0.1:8554/anpviz-camera-dome-comedor
          input_args: preset-rtsp-restream
          roles:
            - record
        - path: rtsp://127.0.0.1:8554/anpviz-camera-dome-comedor_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
    objects:
      track:
        - person
    detect:
      enabled: false 
      width: 1280
      height: 720 
      fps: 5
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 7
    record:
      enabled: True
      events:
        retain:
          default: 14
          mode: active_objects

  anpviz-camera-dome-recibidor:
    ffmpeg:
      hwaccel_args: preset-intel-qsv-h264
      inputs:
      - path: rtsp://192.168.1.32:554/stream0
        roles:
          - record
      - path: rtsp://192.168.1.32:554/stream1
        roles:
          - detect
    objects:
      track:
        - person
    detect:
      enabled: false 
      width: 720
      height: 480 
      fps: 5
    motion:
      mask:
        - 453,170,459,204,482,198,498,182,509,154,518,103,511,82,489,79,469,87
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 7
    record:
      enabled: True
      events:
        retain:
          default: 14
          mode: active_objects

detectors:
  coral:
    type: edgetpu
    device: usb
```

### Relevant log output

```shell
2023-11-28 23:05:40.795955333  [2023-11-29 00:05:40] frigate.video                  ERROR   : anpviz-camera-dome-comedor: Unable to read frames from ffmpeg process.
2023-11-28 23:05:40.796584262  [2023-11-29 00:05:40] frigate.video                  ERROR   : anpviz-camera-dome-comedor: ffmpeg process is not running. exiting capture thread...
2023-11-28 23:05:55.709205065  [2023-11-29 00:05:55] watchdog.anpviz-camera-dome-comedor ERROR   : Ffmpeg process crashed unexpectedly for anpviz-camera-dome-comedor.
2023-11-28 23:05:55.709288568  [2023-11-29 00:05:55] watchdog.anpviz-camera-dome-comedor ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-11-28 23:05:55.709405683  [2023-11-29 00:05:55] ffmpeg.anpviz-camera-dome-comedor.detect ERROR   : rtsp://127.0.0.1:8554/anpviz-camera-dome-comedor_sub: Invalid data found when processing input
2023-11-28 23:05:55.716936923  [2023-11-29 00:05:55] ffmpeg.anpviz-camera-dome-comedor.record ERROR   : rtsp://127.0.0.1:8554/anpviz-camera-dome-comedor: Invalid data found when processing input
2023-11-28 23:05:55.720336730  [2023-11-29 00:05:55] watchdog.anpviz-camera-dome-comedor INFO    : Terminating the existing ffmpeg process...
2023-11-28 23:05:55.720496995  [2023-11-29 00:05:55] watchdog.anpviz-camera-dome-comedor INFO    : Waiting for ffmpeg to exit gracefully...
```

```
2023-11-28 23:06:11.241165815  [2023-11-29 00:06:11] frigate.video                  ERROR   : anpviz-camera-dome-comedor: ffmpeg process is not running. exiting capture thread...
2023-11-28 23:06:15.759231901  [2023-11-29 00:06:15] watchdog.anpviz-camera-dome-comedor ERROR   : Ffmpeg process crashed unexpectedly for anpviz-camera-dome-comedor.
2023-11-28 23:06:15.759547155  [2023-11-29 00:06:15] watchdog.anpviz-camera-dome-comedor ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-11-28 23:06:15.759845658  [2023-11-29 00:06:15] ffmpeg.anpviz-camera-dome-comedor.detect ERROR   : [rtsp @ 0x5596a09113c0] Could not find codec parameters for stream 0 (Video: h264, none): unspecified size
2023-11-28 23:06:15.760372175  [2023-11-29 00:06:15] ffmpeg.anpviz-camera-dome-comedor.detect ERROR   : Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
2023-11-28 23:06:15.760795788  [2023-11-29 00:06:15] ffmpeg.anpviz-camera-dome-comedor.detect ERROR   : Output file #0 does not contain any stream
2023-11-28 23:06:15.781941753  [2023-11-29 00:06:15] ffmpeg.anpviz-camera-dome-comedor.record ERROR   : [rtsp @ 0x55e6a2e01500] Could not find codec parameters for stream 0 (Video: h264, none): unspecified size
2023-11-28 23:06:15.782031525  [2023-11-29 00:06:15] ffmpeg.anpviz-camera-dome-comedor.record ERROR   : Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
2023-11-28 23:06:15.782107196  [2023-11-29 00:06:15] ffmpeg.anpviz-camera-dome-comedor.record ERROR   : Output file #0 does not contain any stream
2023-11-28 23:06:15.782182969  [2023-11-29 00:06:15] watchdog.anpviz-camera-dome-comedor INFO    : Terminating the existing ffmpeg process...
2023-11-28 23:06:15.782275108  [2023-11-29 00:06:15] watchdog.anpviz-camera-dome-comedor INFO    : Waiting for ffmpeg to exit gracefully...
2023-11-28 23:06:20.910626311  [2023-11-29 00:06:20] frigate.video                  ERROR   : anpviz-camera-dome-comedor: Unable to read frames from ffmpeg process.
2023-11-28 23:06:20.910654471  [2023-11-29 00:06:20] frigate.video                  ERROR   : anpviz-camera-dome-comedor: ffmpeg process is not running. exiting capture thread...
```
```


### Frigate stats

_No response_

### Operating system

Debian

### Install method

HassOS Addon

### Coral version

Other

### Any other information that may be helpful

_No response_

**Comment by NickM-27** [comment]
it would be good to see the go2rtc logs

**Comment by marcgarciamarti** [comment]
Hi NickM,

thanks again for helping!

here is the go2rtc log:

```
2023-11-29 20:47:07.591598635  [INFO] Preparing go2rtc config...
2023-11-29 20:47:07.627279914  [INFO] Got IP address from supervisor: 192.168.1.5
2023-11-29 20:47:07.658617529  [INFO] Got WebRTC port from supervisor: 8555
2023-11-29 20:47:07.892111426  [INFO] Starting go2rtc...
2023-11-29 20:47:07.984699372  21:47:07.984 INF go2rtc version 1.2.0 linux/amd64
2023-11-29 20:47:07.985246556  21:47:07.985 INF [api] listen addr=:1984
2023-11-29 20:47:07.985524897  21:47:07.985 INF [rtsp] listen addr=:8554
2023-11-29 20:47:07.985829633  21:47:07.985 INF [srtp] listen addr=:8443
2023-11-29 20:47:07.986066044  21:47:07.986 INF [webrtc] listen addr=:8555
2023-11-29 20:47:13.832570488  21:47:13.832 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:133 > error="read tcp 172.30.33.8:50328->192.168.1.31:554: i/o timeout" url=rtsp://USER:PASS@192.168.1.31:554/stream2
2023-11-29 20:47:13.851332884  21:47:13.850 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:133 > error="read tcp 172.30.33.8:50332->192.168.1.31:554: i/o timeout" url=rtsp://USER:PASS@192.168.1.31:554/stream0
2023-11-29 20:47:17.592564035  [INFO] Starting go2rtc healthcheck service...
2023-11-29 20:47:18.965525563  21:47:18.965 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:133 > error="read tcp 172.30.33.8:57092->192.168.1.31:554: i/o timeout" url=rtsp://USER:PASS@192.168.1.31:554/stream0
2023-11-29 20:47:18.966435522  21:47:18.966 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:133 > error="read tcp 172.30.33.8:57076->192.168.1.31:554: i/o timeout" url=rtsp://guest:passw0rds3gur0@192.168.1.31:554/stream2
2023-11-29 20:47:19.111722910  21:47:19.111 WRN github.com/AlexxIT/go2rtc/cmd/rtsp/rtsp.go:221 > error=EOF
2023-11-29 20:47:19.123734953  21:47:19.123 WRN github.com/AlexxIT/go2rtc/cmd/rtsp/rtsp.go:221 > error=EOF
```

it is strange because it looks like I'm getting a timeout from a URL that looks very much correct... Will do some testing maybe with an admin role instead of guest

**Comment by NickM-27** [comment]
let's try using `preset-vaapi` instead of qsv

**Comment by marcgarciamarti** [comment]
Hello,

thanks Nick for continue to support me. I've done as suggested, still can't get a webrtc feed off the 

This is the only portion that I have altered, from the original configuration

```
  anpviz-camera-dome-comedor:
    ffmpeg:
      hwaccel_args: preset-vaapi
      output_args:
        record: preset-record-generic-audio-copy

      inputs:
        - path: rtsp://127.0.0.1:8554/anpviz-camera-dome-comedor
          input_args: preset-rtsp-restream
          roles:
            - record
        - path: rtsp://127.0.0.1:8554/anpviz-camera-dome-comedor_sub
          roles:
          input_args: preset-rtsp-restream
            - detect
```

this is the go2rtc output:

```
2023-12-01 17:01:41.710728115  [INFO] Preparing go2rtc config...
2023-12-01 17:01:41.762486546  [INFO] Got IP address from supervisor: 192.168.1.5
2023-12-01 17:01:41.793720678  [INFO] Got WebRTC port from supervisor: 8555
2023-12-01 17:01:42.050132178  [INFO] Starting go2rtc...
2023-12-01 17:01:42.129008948  18:01:42.128 INF go2rtc version 1.2.0 linux/amd64
2023-12-01 17:01:42.129311666  18:01:42.129 INF [api] listen addr=:1984
2023-12-01 17:01:42.129807406  18:01:42.129 INF [rtsp] listen addr=:8554
2023-12-01 17:01:42.130122510  18:01:42.130 INF [srtp] listen addr=:8443
2023-12-01 17:01:42.130340423  18:01:42.130 INF [webrtc] listen addr=:8555
2023-12-01 17:01:47.976508548  18:01:47.976 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:133 > error="read tcp 172.30.33.8:43156->192.168.1.31:554: i/o timeout" url=rtsp://192.168.1.31:554/stream2
2023-12-01 17:01:47.976876106  18:01:47.976 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:133 > error="read tcp 172.30.33.8:43162->192.168.1.31:554: i/o timeout" url=rtsp://192.168.1.31:554/stream0
2023-12-01 17:01:51.714587528  [INFO] Starting go2rtc healthcheck service...
2023-12-01 17:01:53.104928288  18:01:53.104 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:133 > error="read tcp 172.30.33.8:43164->192.168.1.31:554: i/o timeout" url=rtsp://192.168.1.31:554/stream2
2023-12-01 17:01:53.105765665  18:01:53.105 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:133 > error="read tcp 172.30.33.8:43176->192.168.1.31:554: i/o timeout" url=rtsp://192.168.1.31:554/stream0
2023-12-01 17:01:53.248151815  18:01:53.247 WRN github.com/AlexxIT/go2rtc/cmd/rtsp/rtsp.go:221 > error=EOF
2023-12-01 17:01:53.252007725  18:01:53.251 WRN github.com/AlexxIT/go2rtc/cmd/rtsp/rtsp.go:221 > error=EOF
2023-12-01 17:02:01.994837199  18:02:01.994 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:133 > error="read tcp 172.30.33.8:55932->192.168.1.31:554: i/o timeout" url=rtsp://192.168.1.31:554/stream0
```

**Comment by marcgarciamarti** [comment]
Hi there! any idea as to what may be happening? I upgraded the firmware on the camera but the issue still occurs. Is there any additional information that may be required to further troubleshoot the issue?

Thanks!

**Comment by marcgarciamarti** [comment]
Hi @NickM-27! Happy new year!!

I thought I could make sense to try the stock go2rtc add-on and it did not make any change. The camera that is not working, stays that way: https://github.com/AlexxIT/go2rtc/issues/860
I contacted the manufacturer and I was provided with an updated firmware. I installed that new firmware but the situaion has not changed either. The camera that does not work was enabling 3 streams (main, sub, third). I tried to disable all of them and just keep the main one. Nothing. Nada. Any idea? I'm at a loss... :(
